src/internal/agent/hooks/flux.go

```diff
@@ -34,15 +36,28 @@ func NewGitRepositoryMutationHook() operations.Hook {
 func mutateGitRepository(r *v1.AdmissionRequest) (*operations.Result, error) {
 	var patches []operations.PatchOperation
 
+	zarfState, err := getZarfStateFromFileWithinAgentPod(zarfStatePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load zarf state from file: %v", err)
+	}
+
+	// Form the gitServerURL from the state
+	gitServerURL := zarfState.GitServer.Address
+	if zarfState.GitServer.Port != 0 {
+		gitServerURL += fmt.Sprintf(":%d", zarfState.GitServer.Port)
```

Comment:
What I said above is not true though when using a non-default port.  If I hosted gittea behind an nginx server and rewrote a subpath to point to my git server then this would result in something like: https://example.com/my-scm:5000...

Reply:
In that case you could add the port as a string within the server URL and just not populate the port flag. If we ever actually need the port we could try to parse the port using `net/url`.

@Racer159 Do you think this works or do you think we need to come up with a more robust solution?

Reply:
At that point why specify the port as a separate thing at all?

Reply:
That's a good point, I thought it would have been needed earlier but I'll just give the url